impeller/entity/geometry/stroke_path_geometry.h

```diff
@@ -65,23 +68,61 @@ class StrokePathGeometry final : public Geometry {
 
   bool SkipRendering() const;
 
-  static Scalar CreateBevelAndGetDirection(
-      VertexBufferBuilder<SolidFillVertexShader::PerVertexData>& vtx_builder,
-      const Point& position,
-      const Point& start_offset,
-      const Point& end_offset);
+  static Scalar CreateBevelAndGetDirection(VertexWriter& vtx_builder,
+                                           const Point& position,
+                                           const Point& start_offset,
+                                           const Point& end_offset);
 
-  static VertexBufferBuilder<SolidFillVertexShader::PerVertexData>
-  CreateSolidStrokeVertices(const Path& path,
-                            Scalar stroke_width,
-                            Scalar scaled_miter_limit,
-                            const JoinProc& join_proc,
-                            const CapProc& cap_proc,
-                            Scalar scale);
+  static void CreateSolidStrokeVertices(VertexWriter& vtx_builder,
+                                        const Path::Polyline& path,
+                                        Scalar stroke_width,
+                                        Scalar scaled_miter_limit,
+                                        const JoinProc& join_proc,
+                                        const CapProc& cap_proc,
+                                        Scalar scale);
 
   static StrokePathGeometry::JoinProc GetJoinProc(Join stroke_join);
 
-  static StrokePathGeometry::CapProc GetCapProc(Cap stroke_cap);
+  static StrokePathGeometry::CapProc GetCapProc(Cap cap);
+
+  static void CreateButtCap(VertexWriter& vtx_builder,
```

Comment:
I'm a bit confused - these don't seem to be used in this PR, is there a reason they were added to the headers?

(these = all the `static void CreateXCap` methods)

Reply:
They are used, see https://github.com/flutter/engine/pull/50379/files#diff-c75818e02ee25c31e8f8a6d5cac9a4f80b33edf3066d39104cd805268310625dR267

Previous this used to pass around closures, but I updated these to be static functions, since that is what they are anyway.